app.py

```diff
@@ -183,24 +184,14 @@ def logout_():
     @authenticated(ADMINS_GROUP)
     @csrf_needed
     def push_log():
-        if 'level' not in request.json or 'message' not in request.json:
-            raise ValueError('Request body missing one or more mandatory fields ["message", "level"]')
+        if type(request.json) != list:
+            raise ValueError('Body must be a list of log entries')
```

Comment:
nitpick: by looking at Python built-in exceptions this should be a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError).

> Passing arguments of the wrong type (e.g. passing a [list](https://docs.python.org/3/library/stdtypes.html#list) when an [int](https://docs.python.org/3/library/functions.html#int) is expected) should result in a [TypeError](https://docs.python.org/3/library/exceptions.html#TypeError), but passing arguments with the wrong value (e.g. a number outside expected boundaries) should result in a [ValueError](https://docs.python.org/3/library/exceptions.html#ValueError).

Reply:
Yes, but that would be appropriate for a library function exception
I chose the ValueError because the "value" received by the API is not correct
Basically I'm using ValueErrors as API validation error